other/upgrade.php

```diff
@@ -3435,19 +3435,22 @@ function serialize_to_json()
 								if ($temp === false)
 									$temp = @safe_unserialize(fix_serialized_data($row[$col]));
 
-								if ($temp === false && $command_line)
+								if ($temp === false)
 								{
-									echo "\nFailed to unserialize " . $row[$col] . "... Skipping\n";
+									if ($command_line)
+										echo "\nFailed to unserialize " . $row[$col] . "... Skipping\n";
+
+									$temp2 = @json_decode($row[$col], true);
+									if (!empty($temp2))
+										$temp = array();
+									elseif ($command_line)
+										echo "... Skipping";
```

Comment:
Checking if the value is already JSON is a good idea. But I think this would be better for this block:
```
$temp = @safe_unserialize($row[$col]);

if ($temp === false)
	$temp = @safe_unserialize(fix_serialized_data($row[$col]));

if ($temp === false)
	$temp = smf_json_decode($row[$col], true, false);

if (empty($temp))
{
	$temp = array();

	if ($command_line)
		echo "\nFailed to unserialize " . $row[$col] . ". Setting to empty value.\n";
}
...
```

Reply:
`fix_serialized_data` never returns `false`

Reply:
No, but `safe_unserialize()` does.